runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -3917,6 +3917,53 @@ J9::CodeGenerator::collectSymRefs(
    }
 
 
+
+void
+J9::CodeGenerator::fixUpProfiledInterfaceGuardTest()
+   {
+   TR::Compilation *comp = self()->comp();
+   TR::CFG * cfg = comp->getFlowGraph();
+   TR::Block *currentBlock = NULL;
+   for (TR::TreeTop * treeTop = comp->getStartTree(); treeTop; treeTop = treeTop->getNextTreeTop())
+      {
+      TR::Node *node = treeTop->getNode();
+
+      if (node->getOpCodeValue() == TR::BBStart )
+         {
+         currentBlock = node->getBlock();
+         }
+      else if (node->getOpCode().isIf() && node->isTheVirtualGuardForAGuardedInlinedCall())
+         {
+         TR_VirtualGuard *vg = comp->findVirtualGuardInfo(node);
+         if (vg && vg->getTestType() == TR_MethodTest && vg->getIsInterfaceMethodGuard())
+            {
+            traceMsg(comp, "Need to add a rangecheck before n%dn in block_%d\n",node->getGlobalIndex(), currentBlock->getNumber());
```

Comment:
this should be a dumpOptDetails rather than just a trace message I would have thought...

Reply:
changed in https://github.com/eclipse/openj9/pull/4721/commits/36292aa9f2fcefeedc9fe76a5f0929a1ce9c2d68